accepted/targeting-packs/targeting-packs.md

```diff
@@ -0,0 +1,149 @@
+  - For the version-specific packages:
+    `Microsoft.NETFramework.ReferencesAssemblies.net462` (where net462 is
+    replaced with the corresponding NuGet short framework Identifier)
+* The NuGet packages will include a `.targets` file that sets the
+  `TargetFrameworkRootPath` to a path within the NuGet package. This will allow
+  the existing MSBuild logic in the `GetReferenceAssemblyPaths` target and other
+  logic such as automatically referencing the facades if necessary to continue
+  to work as normal.
+* The logic in the `.targets` file in the NuGet packages will only set the
+  `TargetFrameworkRootPath` if the `TargetFrameworkIdentifier` and
+  `TargetFrameworkVersion` properties match the reference assemblies that the
+  package provides. It will also only set the property if the
+  `UseReferenceAssembliesFromPackage` property is set to true.
+* The reference assembly packages should include the `.xml` intellisense
+  documentation files. I believe the `.xml` documentation files have to be in
+  the same folder as the reference assemblies, so there isn't a way to deliver
+  them as a separate package.
+* If we want to support localized intellisense, we would need to create a
+  separate set of packages and corresponding meta-package for each language
+  supported. The IDs of these packages could follow the pattern
+  `Microsoft.NETFramework.ReferenceAssemblies.pt-br`. The SDK could use a
+  property to select which language's metapackage to reference.
+* The reference assembly packages should not show up as dependencies of "normal"
+  packages. Thus, the reference assembly packages should set
+  `developmentDependency` to true in it's metadata. Likewise, when the .NET SDK
+  automatically references the reference assembly metapackage, it should use
+  `PrivateAssets="All"`.
+* The reference assembly packages should include the same layout of files that
+  are installed by the targeting packs under `C:\Program Files (x86)\Reference
+  Assemblies\Microsoft\Framework`. This should be rooted at the path specified
+  by the `TargetFrameworkRootPath` property in the package. For example, if the
+  `.targets` file, which is in the `build` folder of the NuGet package, sets the
+  `TargetFrameworkRootPath` to `$(MSBuildThisFileDirectory)`, then the .NET
+  4.6.2 reference assembly package should have the reference assemblies and
+  intellisense files in the `build\.NETFramework\v4.6.2` folder, as well as have
+  the `Facades`, `PermissionSets`, and `RedistList` folders with corresponding
+  files under that folder.
+* The version number of each package should start at 1.0.0. When a new version
+  of the .NET Framework is released, a corresponding reference assembly package
+  (versioned at 1.0.0) should be released, and a new metapackage that includes
+  the additional dependency for the newly supported version should be released.
+  The new version of the metapackage should have it's minor version incremented.
```

Comment:
Ideally, the metapackage would also have a targets file to check for the maximum supported version and report an error if a `net*` version is targeted greater than the currently supported version or else the consuming project might end up with reference assemblies for the lower version if resolution falls back to the latest dependency group of the metapackage.

Reply:
Hey @dsplaisted, is this needed or is this handled by the way the targets work?

Reply:
In my proof of concept, this is basically handled handled by conditioning setting the reference assembly path on the exact target framework version.  The package for 4.7.1, for example, has the following in the .targets file:

```xml
  <PropertyGroup Condition=" ('$(UseReferenceAssembliesFromPackage)' == 'true') And ('$(TargetFrameworkIdentifier)' == '.NETFramework') And ('$(TargetFrameworkVersion)' == 'v4.7.1') ">
    <TargetFrameworkRootPath>$(MSBuildThisFileDirectory)</TargetFrameworkRootPath>
    <_NeedToRemoveExplicitReference>true</_NeedToRemoveExplicitReference>
  </PropertyGroup>
```

So you wouldn't get the wrong reference assemblies, you'd just get the same error you get today if reference assemblies aren't available.

Reply:
We should improve on the error that you get today. It *warns*, "hey, you'll get refs from the GAC, which is totally wrong and stupid, but YOLO the build might succeed." This is especially silly in SDK projects where the GAC is excluded from search paths.

Reply:
(and where there often is no GAC (.NET Core msbuild)